rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -331,6 +336,10 @@ TEST_F(TestExecutor, spin_all_fail_wait_set_clear) {
     node->create_wall_timer(std::chrono::milliseconds(1), [&]() {});
 
   dummy.add_node(node);
+  auto callback = [](rcl_interfaces::msg::ParameterEvent::ConstSharedPtr) {};
+  auto event_sub = rclcpp::AsyncParametersClient::on_parameter_event(
+      node->get_node_topics_interface(),
+      callback);
```

Comment:
@jmachowinski i need your help here... 🙇 since it does not create the `/parameter_events` subscription via node topic interface on node creation internally, it does not hit `rcl_wait_set_clear` and raise the expected exception.

this does not work either.

```suggestion
  auto callback = [this](test_msgs::msg::Empty::ConstSharedPtr) {};
  rclcpp::Subscription<test_msgs::msg::Empty>::SharedPtr subscription;
  auto node_topics = node->get_node_topics_interface();
  subscription =
    rclcpp::create_subscription<test_msgs::msg::Empty>(
      node_topics, "test", rclcpp::QoS(10), std::move(callback));
```

Do you have idea to make this test pass?

Reply:
either waiting longer, or spinning a second time should trigger this. rcl_wait_set_clear should be called, whenever a waitset gets rebuild, and it was not changed in size. Normally this happens after an event was processed, and we are waiting for any further event.